.github/workflows/main.yml

```diff
@@ -1,14 +1,47 @@
 name: main
-on:
-  push:
-    branches:
-      - "**"
+
+# If a workflow is skipped due to path filtering, branch filtering or a commit message, 
+# then checks associated with that workflow will remain in a "Pending" state. 
+# A pull request that requires those checks to be successful will be blocked from merging.
+#
+# So we use conditions to control job execution (`if: ...`) instead of `pull_request` event parameters.
+# A job that is skipped will report its status as "Success".
+# It will not prevent a pull request from merging, even if it is a required check.
+on: [pull_request]
```

Comment:
Running this workflow on pull requests will not be enough:
we still want all tests and sonarcloud to be run on main and dev branches when we merge into them.

Reply:
A pull request must be open to merge into `dev` and `master, so `on: [pull_request]` is enough here.